File: src/mod10_design_pattern/lesson10_flyweight/java/Cat.java
package mod10_design_pattern.lesson10_flyweight.java;

public class Cat implements MagicalCreature{

    private String type;
    private String color;

    public Cat(String type, String color) {
        this.type = type;
        this.color = color;
    }

    @Override
    public void display(String name, String owner) {
        System.out.println("Cat{" +
                "type='" + type + '\'' +
                ", color='" + color + '\'' +
                ", name='" + name + '\'' +
                ", owner='" + owner + '\'' +
                '}');
    }
}

File: src/mod10_design_pattern/lesson10_flyweight/kotlin/Cat.kt
package mod10_design_pattern.lesson10_flyweight.kotlin


class Cat(
    private val type: String,
    private val color: String
) : MagicalCreature {

    override fun display(name: String, owner: String) {
        println("Cat's name is $name, owner is $owner")
    }
}